### PR TITLE
Bug 1966499: Switch Cypress OLM tests to use supported Red Hat operators

### PR DESCRIPTION
**Pull request by dtaylor113**
- OLM single-install test changed from community Couchbase operator to Red Hat CodeReady Workspaces
- OLM global-install test changed from community Portworx operator to Red Hat Service Binding Operator

I replaced the entire contents of `operator-install-global.spec.ts` and `operator-install-single-namespace.spec.ts` so I suggest looking at these files in 'raw format' as the Diff view is hard to follow due to so many changes.

### CODE REVIEW

File: frontend/packages/operator-lifecycle-manager/integration-tests-cypress/views/operator.ts [hunk 1]
```diff
@@ -0,0 +1,144 @@
+import { projectDropdown } from '@console/cypress-integration-tests/views/common';
+import { detailsPage } from '@console/cypress-integration-tests/views/details-page';
+import { listPage } from '@console/cypress-integration-tests/views/list-page';
+import { modal } from '@console/cypress-integration-tests/views/modal';
+import { nav } from '@console/cypress-integration-tests/views/nav';
+
+export const operator = {
+  install: (
```
Comment (TheRealJon) [inline frontend/packages/operator-lifecycle-manager/integration-tests-cypress/views/operator.ts:8]: The install action should be decoupled from the assertion. What if we want to write a test case for an operator that should fail to install?

File: frontend/packages/operator-lifecycle-manager/integration-tests-cypress/views/operator.ts [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+    operatorName: string,
+    operatorHubCardID: string,
+    installToNamespace: string,
+    installToSingleNamespace: boolean = false,
```
Comment (TheRealJon) [inline frontend/packages/operator-lifecycle-manager/integration-tests-cypress/views/operator.ts:12]: This argument isn't really necessary. We can just take 3 args, and if the `installToNamespace` arg is undefined, we assume a global install and use the 'openshift-operators' namespace.

File: frontend/packages/operator-lifecycle-manager/integration-tests-cypress/views/operator.ts [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+      operator.navToDetailsPage(operatorName, installedNamespace);
+      detailsPage.clickPageActionFromDropdown('Uninstall Operator');
+      modal.shouldBeOpened();
+      modal.modalTitleShouldContain('Uninstall Operator?');
+      cy.get('.loading-skeleton--table').should('not.exist');
+    },
+    checkDeleteAllOperands: () =>
+      cy.byTestID('Delete all operand instances for this operator__checkbox').click(),
+  },
+  createInstance: (
```
Comment (TheRealJon) [inline frontend/packages/operator-lifecycle-manager/integration-tests-cypress/views/operator.ts:71]: Separate CRUD action from assertion so that we aren't tied to a specific post-action state.
We should be able to do something like:

```
operator.createOperand(...)
operator.operandExists(...)
```

File: frontend/packages/operator-lifecycle-manager/integration-tests-cypress/views/operator.ts [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+    cy.get('#root_metadata_name')
+      .clear()
+      .type(linkName);
+    cy.byTestID('create-dynamic-form').click();
+    cy.byTestOperandLink(linkName).should('contain', linkName);
+    cy.log(`navigate to the instance "Details" tab`);
+    cy.byTestOperandLink(linkName).click();
+    cy.byTestSectionHeading(`${name} overview`).should('exist');
+  },
+  deleteInstance: (
```
Comment (TheRealJon) [inline frontend/packages/operator-lifecycle-manager/integration-tests-cypress/views/operator.ts:102]: Decouple action/assertion

File: frontend/packages/operator-lifecycle-manager/integration-tests-cypress/views/operator.ts [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+    modal.shouldBeOpened();
+    modal.submit();
+    modal.shouldBeClosed();
+    cy.byTestID('msg-box-title').should('contain', 'No operands found');
+    cy.byTestID('msg-box-detail').should(
+      'contain',
+      'Operands are declarative components used to define the behavior of the application.',
+    );
+  },
+  uninstall: (operatorName: string, installedNamespace: string) => {
```
Comment (TheRealJon) [inline frontend/packages/operator-lifecycle-manager/integration-tests-cypress/views/operator.ts:124]: Decouple action/assertion

**Review by TheRealJon — COMMENTED** [review]
A couple of nits and a request to decouple our CRUD actions from test assertions. It's okay to have assertions for intermediate state (like form inputs), but I think after the submit button is pressed (or whatever interaction symbolizes the execution of the CRUD operation), we should stop there and let each test case decide what state to assert. That gives us more flexibility to use the helpers for testing edge cases or failed states.

File: frontend/packages/operator-lifecycle-manager/integration-tests-cypress/tests/operator-install-global.spec.ts [hunk 6]
```diff
@@ -1,131 +1,57 @@
...
+const testInstance: TestInstanceProps = {
+  name: 'Infinispan Cluster',
+  kind: 'Infinispan',
+  createDropDownID: 'infinispans.infinispan.org',
+  createTitle: 'Create Infinispan',
+  linkName: `example-global-infinispan`,
+  deleteUrl: '/api/kubernetes/apis/infinispan.org',
+};
+
+describe(`Globally installing "${testOperator.name}" operator in ${testOperator.installedNamespace}`, () => {
```
[the remark this answers is not in the document]
Comment (dtaylor113) [reply frontend/packages/operator-lifecycle-manager/integration-tests-cypress/tests/operator-install-global.spec.ts:20]: Global installs do get installed into a namespace, typically `openshift-operators`, right?

File: frontend/packages/operator-lifecycle-manager/integration-tests-cypress/tests/operator-install-global.spec.ts [hunk 7]
```diff
@@ -1,131 +1,57 @@
...
-xdescribe(`Interacting with a global install mode Operator (${operatorName})`, () => {
+import { operator, TestInstanceProps } from '../views/operator';
+
+const testOperator = {
+  name: 'Data Grid',
+  operatorHubCardID: 'datagrid-redhat-operators-openshift-marketplace',
+  installedNamespace: 'openshift-operators',
+};
+
+const testInstance: TestInstanceProps = {
```
[the remark this answers is not in the document]
Comment (dtaylor113) [reply frontend/packages/operator-lifecycle-manager/integration-tests-cypress/tests/operator-install-global.spec.ts:11]: I've seen it refered to both `instances` and `operands` in our code base.

File: frontend/packages/operator-lifecycle-manager/integration-tests-cypress/views/operator.ts (hunk 1 above)
Comment (dtaylor113) [reply frontend/packages/operator-lifecycle-manager/integration-tests-cypress/views/operator.ts:8]: Please see the use of `cy.intercept()` in [#9142](https://github.com/openshift/console/pull/9142/files#diff-888fcd3f03658ad6355c60586b25467f9f1ca3f95c21fc6869bb2b4f3531a2a3R104) to mock failed uninstallation.

I think it would be rather difficult to decouple assertions from commands in this case since we kinda want to test assertions as we go through the many steps of installing an operator or creating an operand instance.   At the moment we have two tests (OLM install to single and global namespaces), it we took out the assertions, they would have to be duplicated in each test suite which uses the operator view.

File: frontend/packages/operator-lifecycle-manager/integration-tests-cypress/views/operator.ts (hunk 2 above)
Comment (dtaylor113) [reply frontend/packages/operator-lifecycle-manager/integration-tests-cypress/views/operator.ts:12]: I did have it that way, but openned it up to be more consistant between single and global installs, and provide maximum flexibility -wasn't sure if it always `openshit-operators` for every Operator (I know it is for RH Data Grid we're using, and probably for others, but don't know if that will always 100% of the time be the case).

File: frontend/packages/operator-lifecycle-manager/integration-tests-cypress/views/operator.ts (hunk 3 above)
Comment (dtaylor113) [reply frontend/packages/operator-lifecycle-manager/integration-tests-cypress/views/operator.ts:71]: We do assertions along-the-way as we do these OLM ops. 
For example for `operator.install()`
1. assert global or single  'installedNamespace` is in the ns dropdown on the install form
2. click Install button
3. assert 'Operator begun installation' messages
4. click 'view installed operator' button
5. assert operator appears in Installed Operators list
6. assert operator's status reaches 'Succeeded'.

I can't really take out/decouple the assertions from this flow.

Comment (TheRealJon) [reply frontend/packages/operator-lifecycle-manager/integration-tests-cypress/views/operator.ts:71]: I just mean for the final resulting state. I think it's okay to assert certain states along the way, but I think it would be better if the `install` helper was agnostic to whether or not the installation should succeed or not.

Comment (TheRealJon) [reply frontend/packages/operator-lifecycle-manager/integration-tests-cypress/views/operator.ts:71]: Same for the create action, it should be agnostic as to whether or not the creation was successful. That should be up to the test case to assert.

File: frontend/packages/operator-lifecycle-manager/integration-tests-cypress/tests/operator-install-global.spec.ts (hunk 7 above)
[the remark this answers is not in the document]
Comment (TheRealJon) [reply frontend/packages/operator-lifecycle-manager/integration-tests-cypress/tests/operator-install-global.spec.ts:11]: The correct term is Operand

File: frontend/packages/operator-lifecycle-manager/integration-tests-cypress/tests/operator-install-global.spec.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (TheRealJon) [reply frontend/packages/operator-lifecycle-manager/integration-tests-cypress/tests/operator-install-global.spec.ts:20]: Yes, but that is generally abstracted from users.

File: frontend/packages/operator-lifecycle-manager/integration-tests-cypress/views/operator.ts (hunk 3 above)
Comment (dtaylor113) [reply frontend/packages/operator-lifecycle-manager/integration-tests-cypress/views/operator.ts:71]: Same for `operator.createInstance/Operand()`:
1. assert initial "No operands found" messages
2. nav to creation form
3. assert correct form title
4. type operand link name
5. click create button
6. assert link name exists in table
7. click on assert link
8. assert operand details page title

File: frontend/packages/operator-lifecycle-manager/integration-tests-cypress/views/operator.ts (hunk 1 above)
Comment (TheRealJon) [reply frontend/packages/operator-lifecycle-manager/integration-tests-cypress/views/operator.ts:8]: Right, you can mock a failure, but you would still need to either create a new helper or just write the test case manually for that. You wouldn't be able to use your `operator.install` helper at all in order to test a failed install. IMO, it would be better if our helpers were less coupled to specific test cases so that they could be reused. If you wrote an install/uninstall helper that is agnostic to success or failure, then it could be reused in the story that you linked.

File: frontend/packages/operator-lifecycle-manager/integration-tests-cypress/views/operator.ts (hunk 3 above)
Comment (dtaylor113) [reply frontend/packages/operator-lifecycle-manager/integration-tests-cypress/views/operator.ts:71]: > I just mean for the final resulting state. I think it's okay to assert certain states along the way...

Ok, I'll more out the last assertions from these commands, and add appropriate `shouldExist` helpers :-)

File: frontend/packages/operator-lifecycle-manager/integration-tests-cypress/views/operator.ts (hunk 2 above)
Comment (TheRealJon) [reply frontend/packages/operator-lifecycle-manager/integration-tests-cypress/views/operator.ts:12]: In current OLM, it will always be 'openshift-operators'. That's a system namespace that's specifically there for globally installed operators.

**Comment by dtaylor113** [comment]
Hi @TheRealJon, I believe I have addressed all your review comments.
Bigger issue might be that we cannot use 'Data Grid' operator, seems to be taking a long time to install during CI -might have to switch to a different Red Hat Operator & Operand.

File: frontend/packages/operator-lifecycle-manager/integration-tests-cypress/views/operator.ts [hunk 8]
```diff
@@ -0,0 +1,164 @@
...
+    // Install
+    cy.byTestID('install-operator').click();
+    cy.log('verify Operator began installation');
+    cy.byTestID('view-installed-operators-btn').should(
+      'contain',
+      `View installed Operators in Namespace`,
+    );
+    cy.log(`navigate to OperatorHub in Namespace ${installToNamespace}`);
+    cy.byTestID('view-installed-operators-btn').click();
+    cy.byLegacyTestID('resource-title').should('contain', 'Installed Operators');
```
Comment (TheRealJon) [inline frontend/packages/operator-lifecycle-manager/integration-tests-cypress/views/operator.ts:42]: This is still coupled with a successful install. The button you are looking for in these last steps will only appear after a successful installation. I would either move these into the test case or make a separate helper that verifies the successful in-progress steps of the installation.
```suggestion
  },
  installCompleted: () =>
    cy.log('verify Operator began installation');
    cy.byTestID('view-installed-operators-btn').should(
      'contain',
      `View installed Operators in Namespace`,
    );
    cy.log(`navigate to OperatorHub in Namespace ${installToNamespace}`);
    cy.byTestID('view-installed-operators-btn').click();
    cy.byLegacyTestID('resource-title').should('contain', 'Installed Operators');
```

**Comment by dtaylor113** [comment]
/retest
network connection issues

File: frontend/packages/operator-lifecycle-manager/integration-tests-cypress/views/operator.ts (hunk 8 above)
Comment (TheRealJon) [reply frontend/packages/operator-lifecycle-manager/integration-tests-cypress/views/operator.ts:42]: My mistake, that link shows immediately.

**Review by TheRealJon — APPROVED** [review]
/lgtm

**Comment by dtaylor113** [comment]
Added WIP label while I limit e2e tests to just cypress OLM testing.

**Comment by dtaylor113** [comment]
/retest
1st test passed!

**Comment by dtaylor113** [comment]
/test ci/prow/e2e-gcp-console
1st test passed!

**Comment by dtaylor113** [comment]
/test e2e-gcp-console
1st test passed

**Comment by dtaylor113** [comment]
/test e2e-gcp-console
2nd test passed

**Comment by dtaylor113** [comment]
@spadgett, @TheRealJon, @rhamilto CI seems to be passing using the new Red Hat CodeReady Workspaces
and Red Hat Service Binding operators.  PTAL -thanks

**Comment by dtaylor113** [comment]
Added WIP label, noticing some flakes :-(  Navigation doesn't seem to happen consistently.

**Review by rhamilto — APPROVED** [review]
LGTM.  Will let @spadgett provide final approval.

File: frontend/packages/operator-lifecycle-manager/integration-tests-cypress/tests/operator-install-single-namespace.spec.ts [hunk 9]
```diff
@@ -1,125 +1,80 @@
...
-  it(`creates a ${operatorName} ${operatorInstance} instance via the form`, () => {
-    cy.log('create a new instance');
-    cy.byTestID('create-dynamic-form').click();
-    cy.byTestOperandLink(operandLink).should('contain', operandLink);
-  });
+    nav.sidenav.clickNavLink(['Operators', 'Installed Operators']);
+    projectDropdown.selectProject(GlobalInstalledNamespace);
+    projectDropdown.shouldContain(GlobalInstalledNamespace);
+    cy.get('.loading-skeleton--table').should('not.exist');
+    // eslint-disable-next-line promise/catch-or-return
```
Comment (spadgett) [inline frontend/packages/operator-lifecycle-manager/integration-tests-cypress/tests/operator-install-single-namespace.spec.ts:61]: We should look at disabling this rule for the integration tests folder.

**Review by spadgett — APPROVED** [review]
/lgtm
